launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -82,8 +83,12 @@
   /** Used internally to create unique logger names. */
   private static final AtomicInteger COUNTER = new AtomicInteger();
 
+  public static final ThreadFactory REDIRECTOR_FACTORY = new NamedThreadFactory("launcher-proc-%d");
+
   static final Map<String, String> launcherConfig = new HashMap<>();
 
+  private OutputRedirector redirector; // Holds the possible OutputRedirector
```

Comment:
There's probably no need to keep this reference. `OutputRedirector` starts a thread so it won't be garbage collected.